myfitnesspal/client.py

```diff
@@ -795,3 +801,75 @@ def get_food_item_details(self, mfp_id: int) -> FoodItem:
             serving_sizes=details["serving_sizes"],
             client=self,
         )
+
+    def get_report(
+        self,
+        report_name: str = "Net Calories",
+        report_category: str = "Nutrition",
+        lower_bound: Optional[datetime.date] = None,
+        upper_bound: Optional[datetime.date] = None,
+    ) -> Dict[datetime.date, float]:
+        """
+        Returns report data of a given name and category between two dates.
+        """
+        if upper_bound is None:
+            upper_bound = datetime.date.today()
+        if lower_bound is None:
+            lower_bound = upper_bound - datetime.timedelta(days=30)
+
+        # If dates are in the opposite order, flip them for convenience
+        if lower_bound > upper_bound:
+            lower_bound, upper_bound = upper_bound, lower_bound
+
+        # Get the URL for the report
+        json_data = self._get_json_for_url(
+            self._get_url_for_report(report_name, report_category, lower_bound)
+        )
+
+        report = OrderedDict(self._get_report_data(json_data))
+
+        if not report:
+            raise ValueError(
+                "Could not load any results for the given category & name"
+            )
+
+        # Remove entries that are not within the dates specified
+        for date in list(report.keys()):
+            if not upper_bound >= date >= lower_bound:
```

Comment:
Does this actually work correctly? I would think that `upper_bound >= date` is first evaluated (to a boolean) and we then go on to compare that boolean to `lowed_bound`

Reply:
I believe it will, fwiw -- Python has special handling for two-sided comparisons like this; eg:

```
>>> not 1 < 2 < 3
False
>>> 1 < 2 < 3
True
```